File: cdk_lib/event_log.py
from aws_cdk import aws_dynamodb, core
from aws_cdk.aws_dynamodb import Attribute


def event_log_table(scope: core.Construct):
    return aws_dynamodb.Table(scope, 'EventLogTable',
                              table_name='event-log',
                              partition_key=Attribute(name='id', type=aws_dynamodb.AttributeType.STRING))

File: cdk/lib/common/event-log.ts
import {Construct, RemovalPolicy} from "@aws-cdk/core"
import {AttributeType, Table} from "@aws-cdk/aws-dynamodb"


export function createEventLogTable(scope: Construct, region: string): Table {
    return new Table(scope, 'EventLogTable', {
        tableName: `event-log-${region}`,
        partitionKey: {name: 'id', type: AttributeType.STRING},
        removalPolicy: RemovalPolicy.DESTROY
    })
}
